File: RedSheet.DbEntities/Models/vEventAgendaTiming.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("vEventAgendaTimings", Schema = "dbo")]
	[Serializable]
	public partial class vEventAgendaTiming  
	{
 
	[DatabaseGenerated(DatabaseGeneratedOption.None)] 
	[Key]
	public int EventAgendaTimingId { get; set; } 
 
	[Required] 
	[MaxLength(100)]
	public string NegotiationPhase { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int NegotiationPhaseId { get; set; } 
 
	[Required] 
	[MaxLength(1000)]
	public string PayOff { get; set; } 
 
	[Required] 
	[MaxLength(1000)]
	public string Process { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int ProjectEventTimelineId { get; set; } 
 
	[Required] 
	[MaxLength(1000)]
	public string Purpose { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int SortOrder { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int TacticsId { get; set; } 
 
	[Required] 
	[MaxLength(200)]
	public string TacticsToBeUsed { get; set; } 
 
	[Required]
	public TimeSpan Time { get; set; } 
 
	[Required] 
	[MaxLength(1000)]
	public string Topic { get; set; } 
 
	[Required] 
	[MaxLength(1000)]
	public string Trigger { get; set; } 

	public vEventAgendaTiming()
	{
    }

    }
}

File: redsheet-portal/src/app/database-models/v-event-agenda-timing.ts
import { required, maxLength, range, nested } from '@rx/annotations';

export class vEventAgendaTiming {
    constructor(vEventAgendaTiming?: vEventAgendaTiming )  {
        let properties = [ "eventAgendaTimingId", "negotiationPhase", "negotiationPhaseId", "payOff", "process", "projectEventTimelineId", "purpose", "sortOrder", "tacticsId", "tacticsToBeUsed", "time", "topic", "trigger",];
        for (let property of properties)
            if (vEventAgendaTiming && vEventAgendaTiming[property])
                this[property] = vEventAgendaTiming[property];
    }
 
	eventAgendaTimingId : number =   0 ;
 
    @required()
    @maxLength(100)
	negotiationPhase : string =   undefined;
 
    @range(1,2147483647)
	negotiationPhaseId : number =   undefined;
 
    @required()
    @maxLength(1000)
	payOff : string =   undefined;
 
    @required()
    @maxLength(1000)
	process : string =   undefined;
 
    @range(1,2147483647)
	projectEventTimelineId : number =   undefined;
 
    @required()
    @maxLength(1000)
	purpose : string =   undefined;
 
    @range(1,2147483647)
	sortOrder : number =   undefined;
 
    @range(1,2147483647)
	tacticsId : number =   undefined;
 
    @required()
    @maxLength(200)
	tacticsToBeUsed : string =   undefined;
 
    @required()
	time : Date =   undefined;
 
    @required()
    @maxLength(1000)
	topic : string =   undefined;
 
    @required()
    @maxLength(1000)
	trigger : string =   undefined;


}
